app/native/ci-build-protos.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# this needs to run on both android (linux) and ios (osx) builds
+
+OS=$(uname -s)
+# Determine the architecture
+ARCH=$(uname -m)
+
+OS_ARCH=""
+# for some reason protoc-gen-grpc-web uses "darwin" instead of "osx"
+OS_ARCH_GRPC_WEB=""
+
+echo "Running on OS=$OS, ARCH=$ARCH"
+
+if [[ "$OS" == "Linux" && "$ARCH" == "x86_64" ]]; then
+    OS_ARCH="linux-x86_64"
+    OS_ARCH_GRPC_WEB="linux-x86_64"
+elif [[ "$OS" == "Darwin" && "$ARCH" == "arm64" ]]; then
+    OS_ARCH="osx-aarch_64"
+    OS_ARCH_GRPC_WEB="darwin-aarch64"
+else
+    echo "Not one of the known builders!"
+    exit 1
+fi
+
+echo "Using OS/arch: $OS_ARCH"
+
```

Comment:
I have no idea what this is so just gonna trust you since it's in the separate native folder lol.